src/ledger/orderbook.ts

```diff
@@ -1,121 +1,48 @@
-  asks: FormattedOrderbookOrder[];
-}
-
-function isSameIssue(a: Issue, b: Issue) {
-  return a.currency === b.currency && a.counterparty === b.counterparty;
-}
+  BookOffer,
+  BookOffersRequest,
+  BookOffersResponse,
+} from "../models/methods/bookOffers";
 
-function directionFilter(direction: string, order: FormattedOrderbookOrder) {
-  return order.specification.direction === direction;
-}
-
-function flipOrder(order: FormattedOrderbookOrder) {
-  const specification = order.specification;
-  const flippedSpecification = {
-    quantity: specification.totalPrice,
-    totalPrice: specification.quantity,
-    direction: specification.direction === "buy" ? "sell" : "buy",
-  };
-  const newSpecification = _.merge({}, specification, flippedSpecification);
-  return _.merge({}, order, { specification: newSpecification });
-}
+import { orderFlags } from "./parse/flags";
 
-function alignOrder(
-  base: Issue,
-  order: FormattedOrderbookOrder
-): FormattedOrderbookOrder {
-  const quantity = order.specification.quantity;
-  return isSameIssue(quantity, base) ? order : flipOrder(order);
+interface Orderbook {
+  buy: BookOffer[];
+  sell: BookOffer[];
 }
 
-export function formatBidsAndAsks(
-  orderbook: OrderbookInfo,
-  offers: BookOffer[]
-) {
-  // the "base" currency is the currency that you are buying or selling
-  // the "counter" is the currency that the "base" is priced in
-  // a "bid"/"ask" is an order to buy/sell the base, respectively
-  // for bids: takerGets = totalPrice = counter, takerPays = quantity = base
-  // for asks: takerGets = quantity = base, takerPays = totalPrice = counter
-  // quality = takerPays / takerGets; price = totalPrice / quantity
-  // for bids: lowest quality => lowest quantity/totalPrice => highest price
-  // for asks: lowest quality => lowest totalPrice/quantity => lowest price
-  // for both bids and asks, lowest quality is closest to mid-market
-  // we sort the orders so that earlier orders are closer to mid-market
-  const orders = offers
-    .sort((a, b) => {
-      const qualityA = a.quality ?? 0;
-      const qualityB = b.quality ?? 0;
-
-      return new BigNumber(qualityA).comparedTo(qualityB);
-    })
-    .map(parseOrderbookOrder);
-
-  const alignedOrders = orders.map(_.partial(alignOrder, orderbook.base));
-  const bids = alignedOrders.filter(_.partial(directionFilter, "buy"));
-  const asks = alignedOrders.filter(_.partial(directionFilter, "sell"));
-  return { bids, asks };
-}
-
-// account is to specify a "perspective", which affects which unfunded offers
-// are returned
-async function makeRequest(
-  client: Client,
-  taker: string,
-  options: GetOrderbookOptions,
-  takerGets: Issue,
-  takerPays: Issue
-) {
-  const orderData = utils.renameCounterpartyToIssuerInOrder({
-    taker_gets: takerGets,
-    taker_pays: takerPays,
-  });
-  return client.requestAll({
+async function getOrderbook(
+  this: Client,
+  taker_pays: Currency,
+  taker_gets: Currency,
```

Comment:
These should be `Amount`s. There will be a tiny bit of extra processing for XRP.